autopaho/auto.go

```diff
@@ -594,7 +594,12 @@ connectionLoop:
 				c.debug.Printf("publishing message from queue with topic %s", pub2.Topic)
 				if _, err = cli.PublishWithOptions(ctx, &pub2, paho.PublishOptions{Method: paho.PublishMethod_AsyncSend}); err != nil {
 					c.errors.Printf("error publishing from queue: %s", err)
-					if errors.Is(err, paho.ErrNetworkErrorAfterStored) { // Message in session so remove from queue
+					if errors.Is(err, paho.ErrInvalidArguments) { // Some errors should not be retried
+						if err := entry.Remove(); err != nil {
+							c.errors.Printf("error removing queue entry: %s", err)
+						}
+						// Need a way to notify the user of this
```

Comment:
I think having a callback for `PublishViaQueue` would be useful so the user can be notified. That same callback could be passed to the call to `PublishWithOptions` if/when callbacks are added for async published in `paho`

Reply:
Agreed - but I think that can be added at a later date (I think this should be done as part of #216; the idea being that the callback should be called whenever the publish is complete whether successfully or stopped due to error).